extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcMetaStoreManagerFactory.java

```diff
@@ -97,12 +98,15 @@ private void initializeForRealm(
     DatasourceOperations databaseOperations = getDatasourceOperations(isBootstrap);
     sessionSupplierMap.put(
         realmContext.getRealmIdentifier(),
-        () ->
-            new JdbcBasePersistenceImpl(
-                databaseOperations,
-                secretsGenerator(realmContext, rootCredentialsSet),
-                storageIntegrationProvider,
-                realmContext.getRealmIdentifier()));
+        new CachedSupplier<>(() ->
+              new JdbcBasePersistenceImpl(
+                  databaseOperations,
+                  secretsGenerator(realmContext, rootCredentialsSet),
+                  storageIntegrationProvider,
+                  realmContext.getRealmIdentifier())));
+
+    // Ensure the supplier caches the first time
+    var unused = sessionSupplierMap.get(realmContext.getRealmIdentifier()).get();
```

Comment:
If your intent is to force the lazy loading, you should probably build that into a `CachedSupplier` method like `CachedSupplier.load` or something. At the very least you don't need to assign a variable.